recipes/wxsqlite3/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get, rmdir, rm
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.premake import Premake, PremakeDeps, PremakeToolchain
+
+required_conan_version = ">=2.19.0"
+
+class WxSqLite3Conan(ConanFile):
+    name = "wxsqlite3"
+    description = "wxSQLite3 is a C++ wrapper around the SQLite database designed for use in wxWidgets applications."
+    license = "LGPL-3.0+ WITH WxWindows-exception-3.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://utelle.github.io/wxsqlite3/docs/html/index.html"
+    topics = ("wxwidgets", "sqlite", "sqlite3", "sql", "database")
+    settings = "os", "compiler", "build_type", "arch"
+    package_type = "library"
+    options = { "shared": [True, False] }
+    options = { "shared": [True, False], "fPIC": [True, False] }
+    default_options = { "shared": False, "fPIC": True }
+    implements = ["auto_shared_fpic"]
+
+    def _arch_to_msbuild_platform(self, arch):
+        platform_map = {
+            "x86": "Win32",
+            "x86_64": "Win64",
+        }
+        platform = platform_map.get(str(arch))
+        return platform
+
+    def _msvc_version_str(self, compiler_version = None):
+        if compiler_version is None:
+            compiler_version = self.settings.compiler.version
+
+        version_map = {
+            "193": "vc17",
+            "194": "vc17",
+        }
+        version = version_map.get(str(compiler_version))
+        return version
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            platform = self._arch_to_msbuild_platform(self.settings.arch)
+            if not platform:
+                raise ConanInvalidConfiguration(f"Unsupported architecture: {self.settings.arch}")
+
+            version = self._msvc_version_str()
+            if not version:
+                raise ConanInvalidConfiguration(f"Unimplemented compiler version: {self.settings.compiler.version}")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def requirements(self):
+        # https://github.com/utelle/wxsqlite3/blob/v4.10.8/premake/wxwidgets.lua#L146
+        self.requires("wxwidgets/[>=3.2.5 <3.3]", transitive_headers=True, transitive_libs=True)
+
+    def build_requirements(self):
+        if self.settings.os == "Windows":
+            self.tool_requires("premake/5.0.0-beta7")
```

Comment:
```suggestion
            self.tool_requires("premake/5.0.0-beta7")
        else:
            self.tool_requires("libtool/2.4.7")
```

Need to add libtool in order to run autoreconf